rover/Process.proto

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+syntax = "proto3";
+
+package skywalking.v3;
+
+option java_multiple_files = true;
+option java_package = "org.apache.skywalking.apm.network.rover.process.v3";
+option go_package = "skywalking.apache.org/repo/goapi/collect/rover/process/v3";
+
+import "common/Common.proto";
+
+// Define the Rover detected processes and report them.
+service ProcessService {
+    // Report discovered process in Rover
+    rpc reportProcesses (ProcessReportList) returns (ReportProcessDownstream) {
+    }
+
+    // Keep the process alive in the backend.
+    rpc keepAlive (ProcessPingPkgList) returns (KeepAliveProcessDownstream) {
+    }
+}
+
+message ProcessReportList {
+    repeated ProcessProperties processes = 1;
+}
+
+message ProcessProperties {
+    // Detect process mode
+    ProcessOwner owner = 1;
+    // The Process metadata
+    oneof metadata {
+        HostProcessMetadata hostProcess = 2;
+        KubernetesMetadata k8sMetadata = 3;
+    }
+}
+
+enum ProcessOwner {
+    KUBERNETES = 0;
+    VM = 1;
+}
+
+message HostProcessMetadata {
+    // [required] Service name of process
+    // Used to logically aggregate multiple Process
+    string serviceName = 1;
+    // [required] User define or generated process name
+    string name = 2;
```

Comment:
How could OAP generate instance name if neither SDK isn't running nor K8s case isn't covered?

Reply:
I think `sha256` should be enough if we use a signing algorithm to identify. 
1. VM-level processes are limited locally.
2. If the Instance Name is too long, more content will be cut when save to the database(service instance id).

Reply:
256 is fine. I just don't like md5😄